Replace STAMPED properties with STAMPED principles in `main.tex` (#125)
myyoda/principles-paper: main.tex

Reviewer: ```suggestion
The Workflow Community Initiative (WCI-FW)\cite{wilkinson_applying_2025} mapped these properties onto computational workflows but explicitly chose not to define new ones, instead treating workflows as hybrid data/software objects.
```

Using the interchange on FAIR first may smooth the experience

Author: I don't know if you still want to apply this change and to where, but it is currently outdated.

Yes, to some degree. That's what the Workflow Community Initiative does. It says nothing about the use of STAMPED as properties/principles interchangeably. Much prefer something more explicit.

Both frameworks cover overlapping concerns.
For example, persistent identifiers serve both Findability and Tracking, but accent different aspects: FAIR emphasizes discovery and interoperability across repositories, while STAMPED emphasizes improving the operational maturity of scientific workflows in day-to-day research practice.

> Many researchers already employ practices that address these operational principles, such as version control, containerized environments, structured project directories, and documented analysis steps.

Reviewer: ```suggestion
Many researchers already employ practices that address these operational properties, such as version control, containerized environments, structured project directories, and documented analysis steps.
```

Yet the community lacks a shared vocabulary for referring to these ideas, evaluating how thoroughly they are applied, and communicating that evaluation to collaborators and reviewers.

> Here we provide that vocabulary: the STAMPED principles (Self-contained, Tracked, Actionable, Modular, Portable, Ephemeral, and Distributable), originating from the YODA Principles\cite{hanke_yoda_2018}, which describe the operational maturity of a research object.

Reviewer: ```suggestion
Here, we provide that vocabulary.
The STAMPED principles articulate seven properties of a research object: Self-contained, Tracked, Actionable, Modular, Portable, Ephemeral, and Distributable.
These originate from both the YODA Principles\cite{hanke_yoda_2018}, as well as VAMP\cite{hanke_what_2023}, both of which describe the operational maturity of a research object.
```

Reviewer 2: not a native speaker, but I think accenting `Here,` with comma is "too much" and I would like partially return prior version while melding with this suggestion, and I think "the" for "STAMPED principles" is also too much since we do specify which principles -- STAMPED. See e.g. https://www.go-fair.org/fair-principles/ -- no "The" for "FAIR principles", but you would use "the principles" when talking about them, hence (and expanding a bit for flow):

```suggestion
Here we provide that vocabulary.
We introduce STAMPED principles which articulate seven properties of a research object: Self-contained, Tracked, Actionable, Modular, Portable, Ephemeral, and Distributable.
These originate from both the YODA Principles\cite{hanke_yoda_2018}, as well as VAMP\cite{hanke_what_2023}, both of which describe the operational maturity of a research object.
```

Reviewer: Grammarly suggested the comma, and it might be right, but I also agree it is a bit much. So maybe we can selectively violate.

How about this re-re-phrasing?

```suggestion
Here we provide that vocabulary by introducing seven principles which articulate the properties of a research object: Self-contained, Tracked, Actionable, Modular, Portable, Ephemeral, and Distributable (STAMPED).
These originate from both the YODA Principles\cite{hanke_yoda_2018}, as well as VAMP\cite{hanke_what_2023}, both of which describe the operational maturity of a research object.
```

Reviewer 2: I still feel we might do another iteration to smooth out into that listing, e.g. (refinement to above):

```suggestion
Here we provide that vocabulary.
We introduce STAMPED principles which articulate how Self-contained, Tracked, Actionable, Modular, Portable, Ephemeral, and Distributable a research object is, thus describing its operational maturity.
These originate from both the YODA Principles\cite{hanke_yoda_2018}, as well as VAMP\cite{hanke_what_2023}.
```

Reviewer 2: I like it somewhere inbetween both of our final ones (but removing `the` in "articulate ~~the~~ properties"), and if we could avoid twice "of a research object".

Reviewer: @yarikoptic I had modified my previous comment mere moments after posting it, but I think you saw my inital version

How about this?

```suggestion
Here we provide that vocabulary by introducing seven principles which articulate how Self-contained, Tracked, Actionable, Modular, Portable, Ephemeral, and Distributable (STAMPED) a research object is, thus describing its operational maturity.
These originate from both the YODA Principles\cite{hanke_yoda_2018}, as well as VAMP\cite{hanke_what_2023}.
```

Reviewer 2: I agree to agree! let's apply (I will)

Rather than a compliance threshold, each STAMPED principle is a spectrum from a practical minimum---often what researchers are already doing---to an aspirational ideal.
A researcher who stores everything under one project directory, version-controls analysis scripts, and documents how to run them is already meeting the practical minimum for Self-containment, Tracking, and Actionability.
As complexity grows, such as for multi-institutional collaborations with heterogeneous workflows spanning platforms enforcing long-term reproducibility, these same principles guide researchers toward more rigorous practices without requiring wholesale adoption of new tools.

> STAMPED provides a vocabulary for describing and a framework for evaluating and incrementally improving the operational principles of research objects toward enhanced rigor, reproducibility, reusability, and efficiency.

Reviewer: ```suggestion
STAMPED provides a vocabulary for describing and a framework for evaluating and incrementally improving the operational properties of research objects toward enhanced rigor, reproducibility, reusability, and efficiency.
```

I think properties goes better with operational in those cases

\begin{table}[htbp]
\centering

> \caption{The STAMPED Principles. Each principle addresses a distinct dimension of research object organization that contributes to reproducibility. Together, they provide a framework for evaluating and improving the reproducibility of computational research.}

Reviewer: ```suggestion
\caption{The STAMPED Principles. Each property addresses a distinct dimension of research object organization that contributes to reproducibility. Together, they provide a framework for evaluating and improving the reproducibility of computational research.}
```

Author: "The STAMPED Principles. Each property addresses ..." I find this change from "principles" to "properties" abrupt.

Reviewer 2: I agree with @candleindark but also I wonder if we could just avoid repeating it somewhat and actually translating from principle to property:

```suggestion
\caption{The STAMPED Principles where each one addresses a distinct property of research object organization that contributes to reproducibility. Together, they provide a framework for evaluating and improving the reproducibility of computational research.}
```

Reviewer 2: this way we avoid introducing one more term "dimension"!

Reviewer 2: overall we have here 6 mentions of dimension:

```
❯ git grep -i dimension main.tex | nl
     1	main.tex:\caption{The STAMPED Principles. Each principle addresses a distinct dimension of research object organization that contributes to reproducibility. Together, they provide a framework for evaluating and improving the reproducibility of computational research.}
     2	main.tex:Actionability applies to every other STAMPED dimension: at the ideal end, each principle is enacted not through documentation alone but through executable specifications (A.2):
     3	main.tex:No single tool addresses all the dimensions of scientific rigor described above, though many tools contribute to one or several properties simultaneously.
     4	main.tex:STAMPED principles are tool-agnostic. For example, any system that moves a principle from documented to executable moves the research object further along the actionability spectrum, but certain tool categories recur across multiple principles; a well-chosen tool can address several dimensions at once, and a thoughtfully assembled toolchain can cover all of them.
     5	main.tex:Stodden\cite{stodden_scientific_2010} articulated legal and social dimensions of sharing computational research, raising concerns central to Distributability~(D) and Tracking~(T).
     6	main.tex:The facts that FAIR required elaboration to accommodate software\cite{barker_introducing_2022} and workflows\cite{goble_fair_2020,wilkinson_applying_2025}, and that independent standards converged on provenance and packaging, confirms that these dimensions are recurrently demanded by research practice.
```

what about making all of them either property or principle?!

Author: I think this kind of abrupt changes can be avoided if we have an explicit statement in the very beginning of the manuscript. The statement can be a variant of the following.

```
In this paper, use the acronym STAMPED to refer to both the principles that a research activity should adhere to and the properties that the representing research object should exhibit.
```

Once that is established, we can just used STAMPED to refer to the hold collection of principles/properties. I.e. Not using "STAMPED principles" or "STAMPED properties" but use just "STAMPED".

For example, the aforementioned wording:

```
The STAMPED Principles. Each property addresses ...
```

can now be just

```
In STAMPED, each property addresses ...
```

T & Tracking & The state and provenance of all components is recorded. \\
A & Actionability & Research object contains machine-actionable information to carry out procedures to obtain or reproduce content. \\
M & Modularity & All modules are independent and composable. \\

> P & Portability & Research object could be moved to different environments while retaining its STAMPED principles. \\

Reviewer 2: nope -- we are talking about object , not principles overall

```suggestion
P & Portability & Research object could be moved to different environments while retaining its STAMPED properties. \\
```

This ensures that nothing is implicitly assumed about the host system, a concern further addressed by Portability.
At the ideal end, every reference is precise enough that retrieval is unambiguous; how to achieve that precision is the concern of Tracking (T) and Distributability (D).

> Self-containment is the foundational principle upon which the remaining STAMPED principles build.

Reviewer: ```suggestion
Self-containment is the foundation upon which the remaining STAMPED principles are built.
```

  \item \textbf{Ephemerality} is more actionable when computation can be orchestrated in disposable environments (e.g., \texttt{docker compose}, Slurm), not just instructed to ``run in a clean environment.''
  \item \textbf{Distributability} is more actionable when a frozen state can be produced and retrieved by others (e.g., containers, \texttt{npm ci}), not just described with pinning instructions.
\end{itemize}

> The principle is tool-agnostic: any system that moves a principle from documented to executable moves the research object further along the actionability spectrum.

Reviewer: ```suggestion
The principle is tool-agnostic: any system that moves a property from documented to executable moves the research object further along the actionability spectrum.
```

Reviewer: FWIW doesn't even have to be one of our own properties here

Distributability also has a circular relationship with the other STAMPED principles: someone else's distribution effort often serves as the starting point for a new research object's self-containment.
Researchers routinely begin by downloading containers, fetching published datasets, and installing released software---modularly composing the distributions of others to assemble their own self-contained research objects.

The spectrum of Distributability begins where FAIR leaves off: publicly accessible components with documented retrieval instructions.
Toward the ideal end, several complementary strategies reduce the risk that external changes will break a research object's reproducibility: bundling dependencies into containers or archives reduces the surface area of components that can independently change or disappear; hosting on archival infrastructure (Zenodo, DANDI, Software Heritage) with content-addressed identifiers allows recipients to verify integrity regardless of source; and mirroring across multiple platforms protects against the failure of any single registry.

\begin{table}[htbp]
\centering

> \caption{Normative statements about STAMPED principles of research objects.}

Reviewer: ```suggestion
\caption{Normative statements about STAMPED principles as they apply to research objects.}
```

Author: An improvement, but I still find it awkward. I don't see how one can apply principles to research objects. 

What about 

```
Normative statements about STAMPED principles as they apply to producing research objects.
```

I feel that "producing" may not be the right verb. Any suggestion for another verb.

Reviewer 2: sure thing needs to be fixed but I would just KISS as we use both as applies best:

```suggestion
\caption{Normative statements about STAMPED properties of research objects.}
```

\subsection{Enabling tools}

> No single tool addresses all the dimensions of scientific rigor described above, though many tools contribute to one or several principles simultaneously.

Reviewer: ```suggestion
No single tool addresses all the dimensions of scientific rigor described above, though many tools contribute to one or several properties simultaneously.
```

Reviewer 2: if just de-dimensionification, placing aside this sentence meaning overall

```suggestion
No single tool addresses all aspects of scientific rigor described above, though many tools contribute to one or several properties simultaneously.
```

but I think, given next sentence we want  to say, and somewhat get away from 'rigor' here since above were were concentrating on STAMPED properties/principles:

```suggestion
No single technology equips a research object with all STAMPED properties described above, though many tools and frameworks contribute to one or several properties simultaneously.
```

Similarly, version control systems such as Git underpin both Tracking (content-addressed identification and change history) and Self-containment (gathering all components under a single root).
In practice, self-containing all data might be undesired or prohibitive due to size or other concerns. Extensions such as git-annex, DataLad, DVC, and Git LFS strike the balance between the two principles by tracking metadata that identifies specific data versions and obtainability information instead, e.g., content checksums and URLs of remote servers that act as a source, and providing actionable interfaces to obtain the data when necessary.

This overlap is not incidental — it reflects the interdependence of the STAMPED principles themselves.
Achieving Ephemerality, for instance, inherently exercises Portability and Actionability: a disposable environment must be reconstituted from explicit specifications, encouraging end-to-end automation.
Likewise, Distributability builds on Self-containment and Tracking: a research object can only be shared in a consistent, retrievable state if its boundary is well-defined and all components are precisely identified.

> The tools that serve multiple principles do so precisely because they operationalize these connections.

Reviewer: ```suggestion
The tools that serve multiple properties do so precisely because they operationalize these connections.
```

Reviewer 2: this serve kicked me below too although there a bit different statement but I do smell duplication already... anyways -- I do not like "serve principle", I would prefer "enhance property", and likely iterate "research object", e.g.

```suggestion
The tools that enhance multiple properties of a research object do so precisely because they operationalize these connections.
```

WDYT?

Table~\ref{tab:stamped-tools} maps some commonly used tools and technologies to each STAMPED principle.
A comprehensive up-to-date review of existing tools is out of scope for this formalization paper and would quickly become outdated.
To fill that niche, we initiated a satellite project with a website to describe tools and examples, and characterize them in terms of STAMPED and FAIR principles [cite https://stamped-principles.github.io/stamped-examples/stamped\_principles/].

> STAMPED principles are tool-agnostic. For example, any system that moves a principle from documented to executable moves the research object further along the actionability spectrum, but certain tool categories recur across multiple principles; a well-chosen tool can address several dimensions at once, and a thoughtfully assembled toolchain can cover all of them.

Reviewer: Note this is duplicated from https://github.com/stamped-principles/stamped-paper/pull/125/changes#diff-f67098301b6788c19af2853a0e5ad776fda05572b1b1224df46c42d1ad323cbbR161

We should deduplicate after this PR

Reviewer 2: unfortunately that link no longer valid, I see two toolchains but not identical use...

```
❯ git grep 'toolchain'
main.tex:STAMPED principles are tool-agnostic. For example, any system that moves a principle from documented to executable moves the research object further along the actionability spectrum, but certain tool categories recur across multiple principles; a well-chosen tool can address several dimensions at once, and a thoughtfully assembled toolchain can cover all of them.
main.tex:Convergence among existing approaches --- cross-domain standards such as W3C PROV\cite{w3c_provenance_working_group_prov-overview_2013} and RO-Crate\cite{peroni_packaging_2022}, tool-level records such as DataLad's\cite{halchenko_datalad_2021} \texttt{datalad run} provenance, and domain-specific specifications such as BIDS-Prov (BEP028)\cite{bids_community_bep028_2025} --- would let STAMPED research objects carry portable provenance across toolchains without lock-in.
```

Reviewer 2: found it -- part of the sentence on moving from documented to executable indeed  is discussed around [line 162 ATM](https://github.com/stamped-principles/stamped-paper/pull/125/changes#diff-f67098301b6788c19af2853a0e5ad776fda05572b1b1224df46c42d1ad323cbbR162)  so we better avoid it explicitly here, what about simplification:

```suggestion
STAMPED principles are tool-agnostic.
A well-chosen tool can improve several properties at once, and a thoughtfully assembled toolchain can cover all of them.
```

The tools that serve multiple principles do so precisely because they operationalize these connections.

Table~\ref{tab:stamped-tools} maps some commonly used tools and technologies to each STAMPED principle.
A comprehensive up-to-date review of existing tools is out of scope for this formalization paper and would quickly become outdated.
To fill that niche, we initiated a satellite project with a website to describe tools and examples, and characterize them in terms of STAMPED and FAIR principles [cite https://examples.stamped-principles.org].

STAMPED principles are tool-agnostic. For example, any system that moves a principle from documented to executable moves the research object further along the actionability spectrum, but certain tool categories recur across multiple principles; a well-chosen tool can address several dimensions at once, and a thoughtfully assembled toolchain can cover all of them.
Researchers should select tools based on their domain, infrastructure, and team familiarity, recognizing that the tools listed here are illustrative rather than prescriptive: what matters is that the underlying principle is satisfied, regardless of which tool achieves it.

> Where multiple tools serve the same STAMPED principle, the choice between them often involves trade-offs between flexibility and simplicity.

Reviewer 2: I do not like 'serve' here although kinda fits 'serve ... principle', since we talk about improving, let's keep that consistent that tools improve or enhance (not really advance if we remove "dimension")

```suggestion
Where multiple tools enhance the same STAMPED property of an object, the choice between them often involves a trade-off between flexibility and simplicity.
```

AI agents are now generating analysis code, selecting methods, and in some systems running entire research cycles end-to-end\cite{boiko_autonomous_2023, lu_ai_2024, mitchener_kosmos_2025}.
Without explicit structure, this automation can produce results whose provenance and reasoning are irrecoverable\cite{messeri_artificial_2024}.

> STAMPED principles already carry most of what is needed --- self-contained workspaces give agents a well-defined boundary, tracked histories make agent contributions auditable and contribute to their ``memory'', and actionable documentation serves humans and machines alike --- but, as with any new methodology or instrumentation entering science, several of these principles need AI-era extensions.

Reviewer: Haven't gotten to this point with the em-dash purge yet. Need to.

Reviewer: (in follow-up)

This closes a loop with the end-to-end ``AI scientist'' systems that opened this section\cite{boiko_autonomous_2023, lu_ai_2024, mitchener_kosmos_2025}: as soon as an agentic pipeline takes a study specification and returns data, code, figures, and a draft manuscript, the specification and every downstream artifact must together satisfy STAMPED --- not only to keep such pipelines efficient to re-run at lower cost, but to make their outputs introspectable.
Tracked and self-contained research objects let a researcher inspect which model, prompt, and decision led to a particular result, implement ``hand over'' between humans and agents alike and let subsequent actor re-enter a prior state and continue from it without rediscovering context.
At the end, human scientists authoring and iterating over the specification of a STAMPED research object would see how their edits propagate through the generated implementation.
Without these principles, the acceleration might be real but the accountability would be lost.

> With STAMPED principles spec-driven agentic research becomes a collaborative transparent artifact between scientists and the AI systems that serves them both.

Reviewer: ```suggestion
Without the STAMPED principles, the acceleration might be real but the accountability would be lost.
Instead, spec-driven agentic research becomes a collaborative, transparent artifact between scientists and the AI systems that support them.
```
